Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -90,18 +99,19 @@ private static string GetDisplayNameForEnumComparison(object o, decimal? v)
         {
             if (o is null || v is null)
             {
-                return "null";
+                return "<null>";
             }
 
-            if (o.GetType().IsEnum)
+            string typePart = o.GetType().Name;
+            string namePart = o.ToString();
+            if (namePart.Contains(",", StringComparison.Ordinal))
```

Comment:
❓ Where does this `,` come from?

Reply:
It's for flags enums. `(MyEnum.One | MyEnum.Two).ToString()` is `"One, Two"`

Reply:
`Enum.GetName` returns null for flags enums.
```cs
[Flags] enum MyEnum { One = 1, Two = 2 }
Enum.GetName(typeof(MyEnum), MyEnum.One|MyEnum.Two)
// null
```

So this was my attempt format such flags enums, but I admit that I was never fully satisfied with it.

See e.g. this test in the PR
```cs
[Fact]
public void When_enum_does_not_have_specified_flag_it_should_fail_with_a_descriptive_message()
{
    // Arrange
    TestEnum someObject = TestEnum.One | TestEnum.Two;

    // Act
    Action act = () => someObject.Should().HaveFlag(TestEnum.Three, "we want to test the failure {0}", "message");

    // Assert
    act.Should().Throw<XunitException>()
        .WithMessage("Expected*have flag TestEnum.Three(4)*because we want to test the failure message*but found TestEnum.{One, Two}(3).");
}
```

Reply:
What about:

`Expected*have flag TestEnum.Three {value: 4}*because we want to test the failure message*but found TestEnum.One|Two {value: 3}.`

Reasoning:
* Usually we use curly braces to denote internals
* Flags enums are combined using the pipe symbol

Reply:
Looks much better!
I've changed the formatting to your suggestions.